awswrangler/data_quality/_create.py

```diff
@@ -0,0 +1,196 @@
+
+
+@apply_configs
+def create_ruleset(
+    name: str,
+    database: str,
+    table: str,
+    df_rules: Optional[pd.DataFrame] = None,
+    dqdl_rules: Optional[str] = None,
+    description: str = "",
+    client_token: Optional[str] = None,
+    boto3_session: Optional[boto3.Session] = None,
+) -> None:
+    """Create Data Quality ruleset.
+
+    Parameters
+    ----------
+    name : str
+        Ruleset name.
+    database : str
+        Glue database name.
+    table : str
+        Glue table name.
+    df_rules : str, optional
+        Data frame with `rule_type`, `parameter`, and `expression` columns.
+    dqdl_rules : str, optional
+        Data Quality Definition Language definition.
+    description : str
+        Ruleset description.
+    client_token : str, optional
+        Random id used for idempotency. Will be automatically generated if not provided.
+    boto3_session : boto3.Session, optional
+        Ruleset description.
+
+    Examples
+    --------
+    >>> import awswrangler as wr
+    >>> import pandas as pd
+    >>>
+    >>> df = pd.DataFrame({"c0": [0, 1, 2], "c1": [0, 1, 2], "c2": [0, 0, 1]})
+    >>> wr.s3.to_parquet(df, path, dataset=True, database="database", table="table")
+    >>> wr.data_quality.create_ruleset(
+    >>>     name="ruleset",
+    >>>     database="database",
+    >>>     table="table",
+    >>>     dqdl_rules="Rules = [ RowCount between 1 and 3 ]",
+    >>>)
+
+    >>> import awswrangler as wr
+    >>> import pandas as pd
+    >>>
+    >>> df = pd.DataFrame({"c0": [0, 1, 2], "c1": [0, 1, 2], "c2": [0, 0, 1]})
+    >>> df_rules = pd.DataFrame({
+    >>>        "rule_type": ["RowCount", "IsComplete", "Uniqueness"],
+    >>>        "parameter": [None, "c0", "c0"],
+    >>>        "expression": ["between 1 and 6", None, "> 0.95"],
+    >>> })
+    >>> wr.s3.to_parquet(df, path, dataset=True, database="database", table="table")
+    >>> wr.data_quality.create_ruleset(
+    >>>     name="ruleset",
+    >>>     database="database",
+    >>>     table="table",
+    >>>     df_rules=df_rules,
+    >>>)
+    """
+    if df_rules is not None and dqdl_rules:
+        raise exceptions.InvalidArgumentCombination("You must pass either ruleset `df_rules` or `dqdl_rules`.")
+
+    client_glue: boto3.client = _utils.client(service_name="glue", session=boto3_session)
+    dqdl_rules = _create_dqdl(df_rules) if df_rules is not None else dqdl_rules
+
+    try:
+        client_glue.create_data_quality_ruleset(
+            Name=name,
+            Description=description,
+            Ruleset=dqdl_rules,
+            TargetTable={
+                "TableName": table,
+                "DatabaseName": database,
+            },
+            ClientToken=client_token if client_token else uuid.uuid4().hex,
+        )
+    except client_glue.exceptions.AlreadyExistsException:
+        raise exceptions.AlreadyExists(f"Ruleset {name} already exists.")
+
+
+@apply_configs
+def evaluate_ruleset(
+    name: str,
```

Comment:
Any reason why we are limiting ourselves to a single ruleset when we could be evaluating multiple at once?

Reply:
No reason, updated, thanks!